### ISSUE TITLE
False positive: scrollable-region-focusable on menus with programmatic keyboard focus

### ISSUE THREAD

**Issue by WilcoFiers** [dequelabs/axe-core#5363]
### Product

axe-core

### Product Version

4.13.0 (develop)

### Latest Version

- [x] I have tested the issue with the latest version of the product

#### Expectation

`scrollable-region-focusable` should pass. The list is keyboard accessible: activating the Products button (or ArrowDown on it) moves focus to the first item, and arrow keys move between items.

#### Actual

axe reports a violation on `#menu` (`Element should have focusable content` / `Element should be focusable`).

#### How to Reproduce

```html
<style>
  #menu {
    max-height: 6em;
    overflow-y: auto;
    width: 12em;
    margin: 0;
    padding: 0;
    border: 1px solid;
    list-style: none;
  }
  #menu button {
    display: block;
    width: 100%;
    text-align: left;
  }
</style>

<button id="menubutton" aria-expanded="true" aria-controls="menu">
  Products
</button>
<ul id="menu" aria-labelledby="menubutton" tabindex="-1">
  <li><button tabindex="-1">Keyboards</button></li>
  <li><button tabindex="-1">Mice</button></li>
  <li><button tabindex="-1">Monitors</button></li>
  <li><button tabindex="-1">Headsets</button></li>
  <li><button tabindex="-1">Webcams</button></li>
  <li><button tabindex="-1">Microphones</button></li>
  <li><button tabindex="-1">Docking stations</button></li>
  <li><button tabindex="-1">Cables</button></li>
</ul>

<script>
  const menubutton = document.getElementById('menubutton');
  const menu = document.getElementById('menu');
  const menuitems = [...menu.querySelectorAll('button')];

  function setExpanded(isExpanded) {
    menubutton.setAttribute('aria-expanded', String(isExpanded));
    menu.hidden = !isExpanded;
  }

  menubutton.addEventListener('click', () => {
    const isExpanded = menubutton.getAttribute('aria-expanded') === 'true';
    setExpanded(!isExpanded);
    if (!isExpanded) {
      menuitems[0].focus();
    }
  });

  menubutton.addEventListener('keydown', event => {
    if (event.key !== 'ArrowDown') {
      return;
    }
    event.preventDefault();
    setExpanded(true);
    menuitems[0].focus();
  });

  menu.addEventListener('keydown', event => {
    const index = menuitems.indexOf(event.target);
    const offset = { ArrowDown: 1, ArrowUp: -1 }[event.key];
    if (index === -1 || !offset) {
      return;
    }
    event.preventDefault();
    menuitems[(index + offset + menuitems.length) % menuitems.length].focus();
  });
</script>
```

#### Additional context

Customer mega menu. Items stay `tabindex="-1"` and are focused by script. `tabindex="-1"` on the `ul` is only there to stop Chrome from focusing the scroll container.
